### PR TITLE
Support Vulkan 1.1.

### LINKED ISSUE #822: Support Vulkan 1.1
As of version 1.5.2, DXVK now requires a Vulkan 1.1 implementation. This means that until MoltenVK supports 1.1, we won't be able to use it with DXVK. I expect that the number of applications requiring Vulkan 1.1 is going to increase in the future as 1.0 ICDs become rarer. Much has been done to move towards this, but there's still much to be done.

Remaining 1.1 tasks:
- [x] `VK_KHR_external_fence` and `VK_KHR_external_fence_capabilities`
- [x] `VK_KHR_external_memory` and `VK_KHR_external_memory_capabilities`
- [x] `VK_KHR_external_semaphore` and `VK_KHR_external_semaphore_capabilities`
- [x] [`VK_KHR_multiview`](/KhronosGroup/MoltenVK/issues/347) - SPIRV-Cross work is already done
- [x] `VK_KHR_sampler_ycbcr_conversion` - SPIRV-Cross work is already done
- [ ] Feature struct for `VK_KHR_shader_draw_parameters`
- [x] Subgroup operations - SPIRV-Cross work is already done
- [ ] Protected memory - Metal exposes no API for this, but we still need to fill in the feature struct, if only to say we don't support it
- [ ] `vkEnumerateInstanceVersion()`

### PR DESCRIPTION
**Pull request by cdavis5e**
These changes add the remaining missing features of Vulkan 1.1.

Fixes #822.

### CODE REVIEW

File: Common/MVKCommonEnvironment.h [hunk 1]
```diff
@@ -75,6 +75,9 @@ extern "C" {
 /** Directive to identify public symbols. */
 #define MVK_PUBLIC_SYMBOL        __attribute__((visibility("default")))
 
+/** Directive to make a public alias of another symbol. */
+#define MVK_PUBLIC_ALIAS(a, t)   asm(".globl _" #a "; _" #a " = _" #t "\n")
```
Comment (billhollings) [inline Common/MVKCommonEnvironment.h:79]: Hmmm...no way to alias C function name symbols without invoking `asm()`?

All function aliasing references I can find are for C++, which I assume is why you couldn't use that? And the `__attribute__((alias))` seems to be available...but it seem to require the declaration to include function params.

I'm amazed you uncovered this technique. Where did you find it?

Comment (billhollings) [inline Common/MVKCommonEnvironment.h:79]: Hmmm...no way to alias C function name symbols without invoking `asm()`?

All function aliasing references I can find are for C++, which I assume is why you couldn't use that? And the `__attribute__((alias))` seems to be available...but it seem to require the declaration to include function params.

I'm amazed you uncovered this technique. Where did you find it?

**Review by billhollings — APPROVED** [review]
Wow! What a milestone accomplishment! Great work!

In this review, I can't anticipate everything that might be impacted for use cases depending on 1.1, but everything looks good at first scan. We'll see how this falls out in practice as we move forward.

Thanks for taking this on, and congratulations! 🥳🎉

File: Common/MVKCommonEnvironment.h (hunk 1 above)
Comment (cdavis5e) [reply Common/MVKCommonEnvironment.h:79]: > Hmmm...no way to alias C function name symbols without invoking `asm()`?
> 
> All function aliasing references I can find are for C++, which I assume is why you couldn't use that? And the `__attribute__((alias))` seems to be available...but it seem to require the declaration to include function params.

The other thing is that `__attribute__((alias))` doesn't work on Darwin. Clang won't let you use it--even though Mach-O has `N_INDR`. Go figure.

`using` only works on functions in a different `namespace`, and it just makes that name available in the current namespace. Defining a variable with `auto` would've defined a *function pointer*, when what I wanted was to make a *symbol* alias.

> I'm amazed you uncovered this technique. Where did you find it?

I came up with it myself.

**Comment by repi** [comment]
Awesome work!

File: Common/MVKCommonEnvironment.h (hunk 1 above)
Comment (billhollings) [reply Common/MVKCommonEnvironment.h:79]: @cdavis5e 

Building for iOS chokes on a number of identical build errors in the _Create Dynamic Library_ build phase:

    Undefined symbols for architecture arm64:
      "_vkBindBufferMemory2KHR", referenced from:
          MVKInstance::initProcAddrs() in libMoltenVK.a(libMoltenVK.a-arm64-master.o)
          ...

which presumably means those symbols don't appear in the built library. I'm wondering if the `asm()` technique doesn't work on iOS either.

Comment (cdavis5e) [reply Common/MVKCommonEnvironment.h:79]: It's the semicolon. On x86, it's a statement separator, but for ARM it's the comment character. I'll make a PR to fix it.

Comment (cdavis5e) [reply Common/MVKCommonEnvironment.h:79]: #1022

**Pull request merged**